blocks/multicolor-accordion/multicolor-accordion.js

```diff
@@ -0,0 +1,53 @@
+export default async function decorate(block) {
+  const accordionItems = block.querySelectorAll(':scope > div > div');
+  accordionItems.forEach((accordionItem) => {
+    const nodes = accordionItem.children;
+
+    const titleText = nodes[0];
+    const rest = Array.prototype.slice.call(nodes, 1);
+
+    const titleDiv = document.createElement('div');
+    const detailsBtn = document.createElement('button');
+    detailsBtn.innerText = 'Details';
+    detailsBtn.className = 'details-button';
+    titleDiv.append(titleText, detailsBtn);
+
+    titleDiv.classList.add('accordion-trigger');
+
+    const content = document.createElement('div');
+    content.classList.add('accordion-content');
+    rest.forEach((elem) => {
+      content.appendChild(elem);
+    });
+
+    const newItem = document.createElement('div');
+    newItem.appendChild(titleDiv);
+    newItem.appendChild(content);
+
+    newItem.classList.add('accordion-item');
+
+    accordionItem.replaceWith(newItem);
+  });
+
+  const triggers = block.querySelectorAll('.accordion-trigger');
```

Comment:
we are just selecting the titleDiv here which we have already above, makes probably sense to directly attach the listener above